aws/resource_aws_sagemaker_image_version.go

```diff
@@ -0,0 +1,124 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/sagemaker/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/sagemaker/waiter"
+)
+
+func resourceAwsSagemakerImageVersion() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSagemakerImageVersionCreate,
+		Read:   resourceAwsSagemakerImageVersionRead,
+		Update: resourceAwsSagemakerImageVersionCreate,
+		Delete: resourceAwsSagemakerImageVersionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"image_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"base_image": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Add `ForceNew: true` as the base image can't be updated. This will allow the `Update: resourceAwsSagemakerImageVersionCreate,` line above to be removed.
This is more like the behaviour of `aws_lambda_layer_version`.

Reply:
I was thinking about this, recreating will not retain history of version meaning an any given moment there can be a single version of the image. on one hand what you propose is more declarative but im not sure how downstream resources that use an image version will react to an image being recreated with same "tag" and image caching in mind.
WDYT?

Reply:
also see https://docs.aws.amazon.com/sagemaker/latest/dg/studio-byoi-sdk-attach-current-domain.html

```
You must delete all the apps in your domain before you update the domain with the new image version. For information about deleting the apps, see Delete an Amazon SageMaker Studio Domain.
```

Reply:
Following IaC principles we should make the image version immutable (i.e. no `Update`).
If apps refer to an existing image version then a completely new Terraform resource for the new image version should be created.

Reply:
Fair enough, will change.

Reply:
done